warehouse/query-core/src/main/java/datawave/core/iterators/DatawaveFieldIndexCachingIteratorJexl.java

```diff
@@ -544,34 +576,37 @@ public void seek(Range r, Collection<ByteSequence> columnFamilies, boolean inclu
             clearRowBasedHdfsBackedSet();
         } else {
             // inside the original range, so potentially need to reposition keyValues
-            if (keys != null) {
+            if (this.keys != null) {
                 Key startKey = r.getStartKey();
                 // decide if keyValues needs to be rebuilt or can be reused
-                if (!keys.hasNext() || (keys.peek().compareTo(startKey) > 0)) {
-                    keys = new CachingIterator<>(threadSafeSet.iterator());
+                if (!this.keys.hasNext() || (this.keys.peek().compareTo(startKey) > 0)) {
+                    this.keys = new CachingIterator<>(this.threadSafeSet.iterator());
                 }
             }
         }
 
         // if we are not sorting UIDs, then determine whether we have a cq and capture the lastFiKey
         Key lastFiKey = null;
-        if (!sortedUIDs && r.getStartKey().getColumnFamily().getLength() > 0 && r.getStartKey().getColumnQualifier().getLength() > 0) {
-            Key startKey = r.getStartKey();
-            String cq = startKey.getColumnQualifier().toString();
-            int fieldnameIndex = cq.indexOf('\0');
-            if (fieldnameIndex >= 0) {
-                String cf = startKey.getColumnFamily().toString();
-                lastFiKey = new Key(startKey.getRow().toString(), "fi\0" + cq.substring(0, fieldnameIndex),
-                                cq.substring(fieldnameIndex + 1) + '\0' + cf + '\0');
+        Key startKey = r.getStartKey();
+        if (!sortedUIDs) {
+            String cq = WaitWindowObserver.removeMarkers(startKey.getColumnQualifier()).toString();
```

Comment:
Can we get a little more inline documentation as to what we are doing when we update the WaitWindowObserver?

Reply:
I can add more documentation as necessary.  However, this line is a static call to strip the YIELD_AT_BEGIN or YIELD_AT_END marker from the colQual (if it exists).  The WaitWindowObserver itself is not affected.